apps/block_scout_web/assets/js/omni.js

```diff
@@ -0,0 +1,162 @@
+// local cache to re-use data
+let txMap = {};
+
+// default api url point to local server
+let url = "http://localhost:8888";
+
+const xchainMarker = () => {
+  const div = document.createElement("div");
+  div.style =
+    "height: inherit;margin-left: 1vw;background-color: #8258cd;padding: 0 0.5vw;border-radius: 1vw;";
+  div.innerHTML =
+    '<img style="height: 1.5vw;display: inline-block;margin-top: -0.1vw" src="/images/favicon-32x32.png"> <div style="margin-top: 0.2vw;display: inline-block;font-weight: bolder;color: white;">X-Chain</div>';
+  return div;
+};
+
+const customXChainDataView = (details) => {
+  const div = document.createElement("div");
+  div.style =
+    "height: inherit;background-color: rgb(130, 88, 205);padding: 2.5vw 1.5vw 0vw 1.5vw;";
+  div.innerHTML = `
+  <div>
+  <img style="height: 2vw; display: inline-block;margin-top: -0.3vw" src="/images/favicon-32x32.png"> 
+  <div style="margin-top: 0.1vw;display: inline-block;font-weight: bolder;color: white;font-size: 1.2vw;padding-left: 0.5vw;">X-Chain Transaction Details</div></div>
+  <pre style="font-size: 1vw;display: block;margin-top: 0.5vw;font-weight: bolder;color: white;padding-left: 1vw;">
+  ${JSON.stringify(details, null, 4)}
```

Comment:
we'll probably want to make this a bit prettier at some point. but it does have a nice software engineer vibe